### PR TITLE
chore: TSTyche usage example

### PR DESCRIPTION
**Pull request by mrazauskas**
Hey there!

Just wanted to draw your attention that `tsd-lite` got [deprecated](https://github.com/mrazauskas/tsd-lite/issues/364).

The architecture of `tsd` was very limited and did not scale well. `jest-runner-tsd` was initially created to test type in Jest repo. I was contributing type tests to the repo for some time. Currently there are around 1200 assertions. At this scale it became painful to debug or add new type features.

These and similar problems inspired me to build TSTyche. I have spending two years researching and experimenting. Now it is ready for use. It is made to scale, to be fast and to last for years.

How exactly? For instance, it has testing helpers like `describe()` or `it()` with `.only`, `.skip`, `.todo` run mode modifiers. These are priceless if you want to focus on one feature.

It ships with `expect` style assertions that can be passed an expression or a type on both side. This means that types can be compared directly without creating intermediate values. The run mode modifiers can be used directly on assertions like `expect.fail`, `expect.only` or `expect.skip`.

Those are here, because you can use TSTyche assertions in your [functional tests](https://tstyche.org/guide/test-runner#functional-tests). This is what I did ion this PR. No need to repeat code.

TSTyche has a lot to offer. It is fast and lightweight (zero dependencies, install size below 200kB).

Currently TSTyche is used by Immutable.js, Jest, Redwood and few more smaller project (including my own packages).

Repo: https://github.com/tstyche/tstyche
Documentation: https://tstyche.org

---

Sorry, I don’t know how to use `nx`. It should somehow run these TSTyche commands:

```shell
# simply runs all selected tests
tstyche

# run all tests using TypeScript 5.0 and currently install version (useful in CI)
tstyche --target 5.0,current
```

### CODE REVIEW

File: packages/api-client/test/user-agent.spec.ts [hunk 1]
```diff
@@ -3,8 +3,8 @@ import {userAgent} from "../src/lib/user-agent";
 
 describe('user-agent', () => {
     it('create user-agent', () => {
-        const userAgent = userAgent();
-        console.log('userAgent', userAgent);
-        expect(userAgent).toBe(userAgent);
+        const agent = userAgent();
```
Comment (mrazauskas) [inline packages/api-client/test/user-agent.spec.ts:6]: This was an error found by TSTyche.

File: tstyche.config.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "$schema": "https://tstyche.org/schemas/config.json",
+  "testFileMatch": ["**/*.spec.ts"]
```
Comment (mrazauskas) [inline tstyche.config.json:3]: It will type check all files matching the pattern and will run any type test if present.

**Comment by Swit-ChristianKim** [comment]
@mrazauskas 
Thank you, mrazauskas.
I'll think about applying tstyche. :) ~~~!

**Comment by mrazauskas** [comment]
Sure. Ping me if anything is unclear. I will close this for now (;

**Pull request closed without merging**